v2/internal/frontend/desktop/linux/frontend.go

```diff
@@ -306,13 +306,19 @@ func (f *Frontend) processRequest(request unsafe.Pointer) {
 		return
 	}
 
-	cContent := C.CString(string(res.Body))
-	defer C.free(unsafe.Pointer(cContent))
+	var cContent unsafe.Pointer
+	bodyLen := len(res.Body)
+	if bodyLen > 0 {
+		cContent = C.malloc(C.ulong(bodyLen))
+		C.memcpy(cContent, unsafe.Pointer(&res.Body[0]), C.size_t(bodyLen))
+	}
+
 	cMimeType := C.CString(res.MimeType)
 	defer C.free(unsafe.Pointer(cMimeType))
-	cLen := C.long(C.strlen(cContent))
+
+	cLen := C.long(len(res.Body))
 	stream := C.g_memory_input_stream_new_from_data(
-		unsafe.Pointer(C.g_strdup(cContent)),
+		cContent,
 		cLen,
 		(*[0]byte)(C.g_free))
```

Comment:
Am wondering if this free might blow up when the body length is 0?

Reply:
Since cContent is now allocated with `C.malloc` wouldn't it be more consistent to use `C.free` here instead of `C.g_free`?